monai/losses/unified_focal_loss.py

```
@@ -39,119 +38,200 @@ def __init__(
 
         if self.to_onehot_y:
             if n_pred_ch == 1:
-                warnings.warn("single channel prediction, `to_onehot_y=True` ignored.")
+                warnings.warn("single channel prediction, `to_onehot_y=True` ignored.", stacklevel=2)
             else:
-                y_true = one_hot(y_true, num_classes=n_pred_ch)
+                if y_true.shape[1] != n_pred_ch:
+                    y_true = one_hot(y_true, num_classes=n_pred_ch)
 
         if y_true.shape != y_pred.shape:
             raise ValueError(f"ground truth has different shape ({y_true.shape}) from input ({y_pred.shape})")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)